microsoft/testsuites/vm_extensions/kvvm_extension.py

```diff
@@ -0,0 +1,220 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+import os
+import random
+
+from assertpy import assert_that
+
+from lisa import (
+    Logger,
+    Node,
+    TestCaseMetadata,
+    TestSuite,
+    TestSuiteMetadata,
+    simple_requirement,
+)
+from lisa.operating_system import BSD
+from lisa.sut_orchestrator.azure.common import (
+    add_system_assign_identity,
+    check_certificate_existence,
+    check_system_status,
+    create_certificate,
+    create_keyvault,
+    delete_certificate,
+    delete_keyvault,
+    get_node_context,
+    rotate_certificates,
+)
+from lisa.sut_orchestrator.azure.features import AzureExtension
+from lisa.sut_orchestrator.azure.platform_ import AzurePlatform
+from lisa.testsuite import TestResult
+
+
+@TestSuiteMetadata(
+    area="vm_extension",
+    category="functional",
+    description="BVT for Azure Key Vault Extension",
+    requirement=simple_requirement(unsupported_os=[]),
+)
+class AzureKeyVaultExtensionBvt(TestSuite):
+    @TestCaseMetadata(
+        description="""
+        The following test case validates the Azure Key Vault Linux
+        Extension while creating the following resources:
+        A resource group
+        A VM
+        A Key Vault
+        Two certificates in the Key Vault
+        Retrieval of the certificate's secrets
+        through SecretClient class from Azure SDK.
+        Installation of the Azure Key Vault Linux Extension on the VM.
+        Rotation of the certificates (After KVVM Extension has been installed)
+        All of the resources have been created by using the Azure SDK Python.
+        """,
+        priority=0,
+        requirement=simple_requirement(
+            supported_features=[AzureExtension], unsupported_os=[BSD]
+        ),
+    )
+    def verify_key_vault_extension(
+        self, log: Logger, node: Node, result: TestResult
+    ) -> None:
+        # Section for vault name and supported OS check
+        vault_name = f"python-keyvault-{random.randint(1, 100000):05}prp"
+
+        # Section for environment setup
+        environment = result.environment
+        assert environment, "fail to get environment from testresult"
+        platform = environment.platform
+        assert isinstance(platform, AzurePlatform)
+        # VM attributes
+        node_context = get_node_context(node)
+
+        # User's attributes
+        user_tenant_id = os.environ.get("tenant_id")
+        if user_tenant_id is None:
+            raise ValueError("Environment variable 'tenant_id' is not set.")
+        user_object_id = os.environ.get("user_object_id")
+        if user_tenant_id is None:
+            raise ValueError("Environment variable 'user_object_id' is not set.")
+        assert user_object_id is not None
```

Comment:
when tenant id and app id specified in runbook, we can use below os environment.

tenant_id = os.environ["AZURE_TENANT_ID"]
object_id = os.environ["AZURE_CLIENT_ID"]